### PR TITLE
[Page Actions] Re-render without measuring when actions / groups change

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes: https://github.com/Shopify/polaris-react/issues/3575
And allows actions and groups to re-render in their previously assigned position, only re-measuring if the lengths of actions and groups changes

## 

### How to 🎩

There's now a toggle in the details page under more actions that changes the actions length. Make sure that works properly and re-measures on toggle

Also make sure that the view button logs out the updated product description

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by kyledurand** [comment]
This partially addresses https://github.com/Shopify/polaris-ux/issues/522. There might be something more integration-y we can do as well and we'll be exploring that next week

File: src/components/ActionMenu/components/Actions/Actions.tsx [hunk 1]
```diff
@@ -126,8 +159,17 @@ export function Actions({actions = [], groups = []}: Props) {
     }
 
     availableWidthRef.current = actionsLayoutRef.current.offsetWidth;
+    if (
+      // Allow measuring twice
+      // This accounts for the initial paint and re-flow
```
Comment (alex-page) [inline src/components/ActionMenu/components/Actions/Actions.tsx:164]: ✨

**Review by alex-page — APPROVED** [review]
Went through this with @kyledurand, looking good.

**Comment by dfmcphee** [comment]
This looks good. I did another 🎩 here and in web and all looks fine now. 👍

**Comment by kyledurand** [comment]
Added an issue to re-think tests for layout thrashing https://github.com/Shopify/polaris-react/issues/3800 and potentially the codecov things but they're mainly covered in other tests

**Pull request merged**